src/odemis/driver/test/tfsbc_test.py

```diff
@@ -60,6 +67,23 @@ def setUpClass(cls):
     def tearDownClass(cls):
         pass
 
+    def test_dependency_scanner(self):
+        """Test dependency on scanner."""
+        # TODO only works with xt simulator now - how to make generic? for xt simulator has special flag "sim"
+        if TEST_NOHW:
+            self.microscope = xt_client.SEM(**CONFIG_MB_SEM)
+            for child in self.microscope.children.value:
+                if child.name == CONFIG_MB_SCANNER["name"]:
+                    self.scanner = child
+
+            dependencies = {"scanner": self.scanner}
+            self.bc = tfsbc.BeamShiftController("DC Offset", None, PORT, "FT2OMDD5", dependencies)
```

Comment:
why use `self.microscope` and `self.bc`, when you only use it in this test case?

Reply:
Ah this was because I tried to add it directly in the setupclass method. However, that breaks all the other test cases as those rely on the calibration matrix that is hardcoded there. For the simulator we only get zeros returned for the calibration matrix. I think it is not the most elegant solution now, to instantiate the beamshift component in the setupclass and then here again, just with adding some dependency. Any suggestions how I can do this differently? The only idea I had was to create a new test class, but that seems also not right.

Reply:
:thinking: I don't have a good idea right away, a new test class also seems a bit over the top, but I also cannot think of another way

Reply:
@pieleric do you have a good suggestions?